prime-dotnet-webapi/Controllers/EnrolleesController.cs

```diff
@@ -453,5 +451,36 @@ public async Task<ActionResult<EnrolmentCertificateNote>> UpdateEnrolmentCertNot
 
             return Ok(new ApiOkResponse<IEnrolleeNote>(updatedNote));
         }
+
+        // GET: api/Enrollees/5/terms-of-access
+        /// <summary>
+        /// Get the enrolmee's terms of access.
+        /// </summary>
+        /// <param name="enrolleeId"></param>
+        [HttpGet("{enrolleeId}/terms-of-access", Name = nameof(GetTermsOfAccess))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiOkResponse<TermsOfAccess>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<TermsOfAccess>> GetTermsOfAccess(int enrolleeId)
+        {
+            if (!await _enrolleeService.EnrolleeExistsAsync(enrolleeId))
+            {
+                return NotFound(new ApiResponse(404, $"Enrollee not found with id {enrolleeId}"));
+            }
+
+            // Prevent access to the enrollee's current terms of access based on status
+            var statuses = new[] { Status.IN_PROGRESS_CODE, Status.DECLINED_CODE, Status.DECLINED_TOS_CODE };
+            if (await _enrolleeService.IsEnrolleeInStatusAsync(enrolleeId, statuses))
```

Comment:
If the above params suggestion is implemented, this line can change to `if (await _enrolleeService.IsEnrolleeInStatusAsync(enrolleeId, Status.IN_PROGRESS_CODE, Status.DECLINED_CODE, Status.DECLINED_TOS_CODE))` if you want. Its longer but doesn't involve creating an unnecessary array first. Your call.

Reply:
Updated and added as params